### PR TITLE
UI: Redesign status bar

### PR DESCRIPTION
**Pull request by cg2121**
### Description
Makes status bar look better

Screenshot from 2023-05-27 03-18-09

### Motivation and Context
Based on mockup by @GeorgesStavracas. The reason for this change is to better group similar items together in the status bar, and make it look better. For example, before, the network square was all the way to the right, next to the recording timer, when it should of been next to the streaming timer.

This is not 100% complete, a few minor things still need to be done
- [x] The streaming inactive/status inactive icons need a version for light themes
- [x] So far, only themed with Yami
- [x] Needs new recording icons

### How Has This Been Tested?
Ran stream/recorded to make sure everything works

### Types of changes
- Tweak (non-breaking change to improve existing functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by GeorgesStavracas** [comment]
I've pushed various changes. There are now light and dark variants of various inactive icons, but there is no mechanism to switch between them depending on the theme. This should be implemented before landing this PR.

**Comment by cg2121** [comment]
The icons are now updated based on theme. I also went through and optimized all of the svg files.

**Review by GeorgesStavracas — APPROVED** [review]
I cannot comment too much on the Qt code itself, so I'll assume it should be mostly fine. I've tested this and it looks great here! Thank you very much

File: UI/window-basic-status-bar.hpp
Comment (tytan652) [inline UI/window-basic-status-bar.hpp:22]:
```suggestion
	inline ~StatusBarWidget(){};
```

File: UI/window-basic-status-bar.cpp
Comment (tytan652) [inline UI/window-basic-status-bar.cpp:26]:
```suggestion
```

**Review by tytan652 — CHANGES_REQUESTED** [review]
If the destructor is empty, just inline it in the header to avoid writing an empty function.

File: UI/window-basic-status-bar.hpp
Comment (cg2121) [reply UI/window-basic-status-bar.hpp:22]: The destructor needs to be in the cpp file for the std::unique_ptr to work.

**Review by Warchamp7 — APPROVED** [review]

File: UI/window-basic-status-bar.cpp
Comment (notr1ch) [inline UI/window-basic-status-bar.cpp:250]: What is the point of showing the selected output FPS in the status bar? This is a static value once set, so doesn't really belong in a status bar IMO. The current FPS will also match the output FPS in the majority of cases so this looks confusing to the user ("why do I have two FPS numbers?").

Also this seems to be missing on the main screenshot of the PR - UI changes should always have accurate screenshots.

**Comment by notr1ch** [comment]
The alignment of the icons looks slightly off in Dark (didn't test other themes).

align

File: UI/window-basic-status-bar.cpp
Comment (notr1ch) [inline UI/window-basic-status-bar.cpp:18]: We should prefer `constexpr` over macros for new code.

File: UI/window-basic-status-bar.cpp
Comment (notr1ch) [inline UI/window-basic-status-bar.cpp:368]: I understand the desire to avoid rapid state changes, but taking up to 10 seconds for the UI to notify about connection issues feels way too long. Congestion typically comes in bursts, so a user will see dropped frames increasing but the congestion icon will still show an excellent connection until the next update which doesn't seem right.

File: UI/window-basic-status-bar.cpp
Comment (cg2121) [reply UI/window-basic-status-bar.cpp]: Updated to use constexpr

File: UI/window-basic-status-bar.cpp
Comment (cg2121) [reply UI/window-basic-status-bar.cpp]: Updated it to 4 seconds, which seems more reasonable.

**Comment by cg2121** [comment]
The alignment issues have now been fixed with the latest update of the code.

File: UI/window-basic-status-bar.cpp
Comment (cg2121) [reply UI/window-basic-status-bar.cpp:249]: The screenshot has now been updated.

**Comment by WizardCM** [comment]
I finally had a chance to experiment with this a bit, and have a couple of minor pieces of feedback.

The first is that I worry about elements jumping around too much when numbers change. Such as
* When active FPS changes
* When CPU usage changes

I assume this is why the kbps counter is to the left of the list of elements. What happens when bitrate drops from 1000 to 911? Will the "Dropped Frames" counter shift too?

You can see this behaviour here with CPU usage:


https://github.com/obsproject/obs-studio/assets/941350/22411367-a284-4e39-a824-0185b3c681e4

Imagine CPU usage repeatedly jumping between 9% and 10% for example. In all cases, setting a slightly wider minimum width would do the trick. Though this will depend on fonts - I know macOS has a habit of rendering fonts slightly thicker/bolder.

Secondly, I have minor concerns about re-arranging items that we've had for over 10 years without clear reasoning for the user. We've often been hurt by PR "Motivation & Context" being very barebones and unclear. This is one such example. I personally know a lot of effort goes into Georges' mockups, but we need to verbalise the reasoning in the PR and in the release notes.

File: UI/window-basic-status-bar.cpp
Comment (GeorgesStavracas) [reply UI/window-basic-status-bar.cpp:249]: We went with `current / max FPS` after Fenrir suggested switching to `render time / max render time ms` upon clicking. I think it's good to maintain the direct link between both. This behaviour is not implemented in this PR, but eventually it will.

**Comment by Warchamp7** [comment]
The intent here is grouping related information together. Let's get that added to Motivation as Matt mentioned for when patch notes time comes along.

As a proposal for the size changes, I'd suggest checking the size of the widget after each update. If it is larger, set that as the minimum size of the widget. Those minimum sizes should then get reset whenever that output is stopped.

This should mean sizes only "jump" up a very limited number of times and stay at that size to potentially accommodate that jump again later.

**Comment by cg2121** [comment]
Updated to prevent labels from jumping around

File: UI/data/themes/Acri.qss
Comment (RytoEX) [inline UI/data/themes/Acri.qss:340]:
```suggestion
/* Qt enforces a padding inside its status bar, so we
```

File: UI/data/themes/Dark.qss
Comment (RytoEX) [inline UI/data/themes/Dark.qss:647]:
```suggestion
/* Qt enforces a padding inside its status bar, so we
```

File: UI/data/themes/Grey.qss
Comment (RytoEX) [inline UI/data/themes/Grey.qss:340]:
```suggestion
/* Qt enforces a padding inside its status bar, so we
```

File: UI/data/themes/Light.qss
Comment (RytoEX) [inline UI/data/themes/Light.qss:340]:
```suggestion
/* Qt enforces a padding inside its status bar, so we
```

File: UI/data/themes/Rachni.qss
Comment (RytoEX) [inline UI/data/themes/Rachni.qss:342]:
```suggestion
/* Qt enforces a padding inside its status bar, so we
```

File: UI/data/themes/System.qss
Comment (RytoEX) [inline UI/data/themes/System.qss:406]:
```suggestion
/* Status Bar */
```

File: UI/data/themes/Yami.qss
Comment (RytoEX) [inline UI/data/themes/Yami.qss:343]:
```suggestion
/* Qt enforces a padding inside its status bar, so we
```

File: UI/window-basic-status-bar.hpp
Comment (RytoEX) [inline UI/window-basic-status-bar.hpp:10]: Is QLabel still used here? I don't see any class members of this type.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Mostly nits. One or two actual code questions. May take another deeper look early next week.

File: UI/window-basic-status-bar.hpp
Comment (cg2121) [reply UI/window-basic-status-bar.hpp]: Fixed

**Review by Warchamp7 — APPROVED** [review]

File: UI/data/themes/Yami.qss
Comment (RytoEX) [inline UI/data/themes/Yami.qss:138]: This extra line appears in every theme file that used the same base, and it is not the only double blank line before a comment header/label. Might as well leave it be.

**Review by RytoEX — CHANGES_REQUESTED** [review]
One nit.

**Comment by PatTheMav** [comment]
> The intent here is grouping related information together. Let's get that added to Motivation as Matt mentioned for when patch notes time comes along.
> 
> As a proposal for the size changes, I'd suggest checking the size of the widget after each update. If it is larger, set that as the minimum size of the widget. Those minimum sizes should then get reset whenever that output is stopped.
> 
> This should mean sizes only "jump" up a very limited number of times and stay at that size to potentially accommodate that jump again later.

Doesn't need to be addressed in this PR necessarily, but given that `CPU` is not a localised label, the maximum width of the widget is naturally limited:

`CPU: 000.0%`

As such a fixed width that should work? (The absolute "correct" way would be to have a fixed width widget, which contains the left-aligned label, and the right-aligned value enclosed).

**Comment by mufunyo** [comment]
> `CPU` is not a localised label

Shouldn't it be? I can imagine a few languages like Quebecois having a different abbreviation for it.

**Comment by PatTheMav** [comment]
> > `CPU` is not a localised label
> 
> Shouldn't it be? I can imagine a few languages like Quebecois having a different abbreviation for it.

Maybe? But it isn't right now and the PR doesn't change it:

https://github.com/obsproject/obs-studio/blob/fcf0d377195a39ef4062b88506bb90e6ebe88de9/UI/window-basic-status-bar.cpp#L225

File: UI/window-basic-status-bar.cpp
Comment (gxalpha) [inline UI/window-basic-status-bar.cpp:239]: This can't ever be `NULL` I don't think.
Either way, the variable is unused here and can get removed.

File: UI/window-basic-status-bar.cpp
Comment (gxalpha) [inline UI/window-basic-status-bar.cpp:249]: It could be worth doing `QString::asprintf` instead to avoid all the appending.

**Review by sebastian-s-beckmann — COMMENTED** [review]
Wouldn't block this PR on it but the notification text looks misaligned here, should probably be centered:
image
Other minor notes below.

File: UI/data/themes/Yami.qss
Comment (RytoEX) [reply UI/data/themes/Yami.qss]: I've force-pushed the change for this nit.

**Comment by RytoEX** [comment]
I've force pushed the change to resolve my nit and also rebased the PR.

**Comment by RytoEX** [comment]
I've force-pushed a change to resolve the clang-format failure and rebase the PR again. CI should hopefully be green now.

**Review by RytoEX — APPROVED** [review]
Approving to clear my previous "changes requested" status. My nits have been addressed.

**Pull request merged**
